src/coreclr/vm/debugdebugger.cpp

```diff
@@ -230,43 +226,221 @@ FCIMPL0(FC_BOOL_RET, DebugDebugger::IsLogging)
+
+        if (!pTemp)
+        {
+            return SWA_ABORT;
+        }
+
+        memcpy(pTemp, pData->pElements, pData->cElementsAllocated * sizeof(DebugStackTrace::Element));
+
+        delete [] pData->pElements;
+
+        pData->pElements = pTemp;
+        pData->cElementsAllocated *= 2;
+    }
+
+    PCODE ip;
+    DWORD dwNativeOffset;
+
+    if (pCf->IsFrameless())
+    {
+        // Real method with jitted code.
+        dwNativeOffset = pCf->GetRelOffset();
+        ip = GetControlPC(pCf->GetRegisterSet());
+    }
+    else
+    {
+        ip = (PCODE)NULL;
+        dwNativeOffset = 0;
+    }
+
+    // Pass on to InitPass2 that the IP has already been adjusted (decremented by 1)
+    INT flags = pCf->IsIPadjusted() ? STEF_IP_ADJUSTED : 0;
+
+    pData->pElements[pData->cElements].InitPass1(
+            dwNativeOffset,
+            pFunc,
+            ip,
+            flags);
+
+    // We'll init the IL offsets outside the TSL lock.
+
+    ++pData->cElements;
+
+    // Since we may be asynchronously walking another thread's stack,
+    // check (frequently) for stack-buffer-overrun corruptions after
+    // any long operation
+    pCf->CheckGSCookies();
+
+    // check if we already have the number of frames that the user had asked for
+    if ((pData->NumFramesRequested != 0) && (pData->NumFramesRequested <= pData->cElements))
+    {
+        return SWA_ABORT;
+    }
+
+    return SWA_CONTINUE;
+}
+
+static void GetStackFrames(
+    Frame *pStartFrame,
+    void* pStopStack,
+    DebugStackTrace::GetStackFramesData *pData)
 {
     CONTRACTL
     {
-        FCALL_CHECK;
-        PRECONDITION(CheckPointer(pStackFrameHelperUNSAFE));
-        PRECONDITION(CheckPointer(pExceptionUNSAFE, NULL_OK));
+        MODE_COOPERATIVE;
+        GC_TRIGGERS;
+        THROWS;
     }
     CONTRACTL_END;
 
-    STACKFRAMEHELPERREF pStackFrameHelper   = (STACKFRAMEHELPERREF)ObjectToOBJECTREF(pStackFrameHelperUNSAFE);
-    OBJECTREF           pException          = ObjectToOBJECTREF(pExceptionUNSAFE);
-    PTRARRAYREF         dynamicMethodArrayOrig = NULL;
+    ASSERT (pData != NULL);
+
+    pData->cElements = 0;
+
+    // if the caller specified (< 20) frames are required, then allocate
+    // only that many
+    if ((pData->NumFramesRequested > 0) && (pData->NumFramesRequested < 20))
+    {
+        pData->cElementsAllocated = pData->NumFramesRequested;
+    }
+    else
+    {
+        pData->cElementsAllocated = 20;
+    }
+
+    // Allocate memory for the initial 'n' frames
+    pData->pElements = new DebugStackTrace::Element[pData->cElementsAllocated];
+
+    if (pData->TargetThread == NULL ||
+        pData->TargetThread->GetInternal() == GetThread())
+    {
+        // Null target thread specifies current thread.
+        GetThread()->StackWalkFrames(GetStackFramesCallback, pData, FUNCTIONSONLY | QUICKUNWIND, pStartFrame);
+    }
+    else
+    {
+        Thread *pThread = pData->TargetThread->GetInternal();
```

Comment:
This path looks like a dead code. As far as I can tell, `TargetThread` is always set to null.

Reply:
Agreed. I will clean this up.